### PR TITLE
helm: add envoy gateway addon helm chart support

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

ref https://github.com/envoyproxy/gateway/issues/3355

### CODE REVIEW

File: charts/dashboard-addon/dashboards/envoy-clusters.json
Comment (shawnh2) [inline charts/dashboard-addon/dashboards/envoy-clusters.json:1]: do we need to remove https://github.com/envoyproxy/gateway/tree/main/examples/grafana/dashboards ?

File: charts/dashboard-addon/Chart.yaml
Comment (zirain) [inline charts/dashboard-addon/Chart.yaml:2]: gateway-addon would be better?

Comment (arkodg) [reply charts/dashboard-addon/Chart.yaml:2]: +1 to `envoy gateway add-ons`

Comment (shawnh2) [reply charts/dashboard-addon/Chart.yaml:2]: SGTM

File: charts/dashboard-addon/Chart.lock
Comment (shawnh2) [reply charts/dashboard-addon/Chart.lock:8]: of course

**Comment by shawnh2** [comment]
This PR only include DP metrics and Grafana support, will send another PR to include CP metrics as well (https://github.com/envoyproxy/gateway/issues/3474#issuecomment-2132817429)

File: charts/dashboard-addon/Chart.yaml
Comment (shawnh2) [reply charts/dashboard-addon/Chart.yaml]: changed

File: charts/dashboard-addon/dashboards/envoy-clusters.json
Comment (shawnh2) [reply charts/dashboard-addon/dashboards/envoy-clusters.json:1]: > do we need to remove https://github.com/envoyproxy/gateway/tree/main/examples/grafana and https://github.com/envoyproxy/gateway/tree/main/examples/prometheus from examples ?

Comment (shawnh2) [reply charts/dashboard-addon/dashboards/envoy-clusters.json:1]: and update the makefile tools?

Comment (arkodg) [reply charts/dashboard-addon/dashboards/envoy-clusters.json:1]: yeah having one source of truth makes a lot of sense, thanks !

File: charts/gateway-addon/Chart.yaml
Comment (arkodg) [inline charts/gateway-addon/Chart.yaml:1]: the current dir structure is `gateway-helm` , can we rename the dir here to `gateway-add-ons-helm` which would also represent the chart name in the OCI registry.
reg naming, I'll bring this up in the community meeting today

**Comment by arkodg** [comment]
hey @shawnh2 , @zhaohuabing @guydc and I brought this up in the community meeting today and aligned on
* calling this chart `gateway-demo-helm` 
* adding envoy gateway as a sub chart (sorry for the churn here)

**Comment by shawnh2** [comment]
> This PR only include DP metrics and Grafana support, will send another PR to include CP metrics as well ([#3474 (comment)](https://github.com/envoyproxy/gateway/issues/3474#issuecomment-2132817429))

lets include Grafana dashboard for CP as well after https://github.com/envoyproxy/gateway/pull/3506 landed. 

and the doc under https://gateway.envoyproxy.io/latest/tasks/observability/ can be updated with this helm demo support.

**Comment by shawnh2** [comment]
> hey @shawnh2 , @zhaohuabing @guydc and I brought this up in the community meeting today and aligned on
> 
> * calling this chart `gateway-demo-helm`
> * adding envoy gateway as a sub chart (sorry for the churn here)

If we are including `envoy-gateway` helm as well, we may not install this demo chart just using one `helm install`, cause eg and prom/grafana should be installed at different namespace. But helm does not support installing dependencies at different namespace. xref: https://github.com/helm/helm/issues/10905

So:

- Do we accpet installing all the dependencies at the same one namespace?
- There's another tool [helmfile](https://helmfile.readthedocs.io/en/latest/) could solve this problem, but will introduce more operations for users.
- I can understand why we are trying to include eg as a dependency, to provide users with a one-stop eg-with-addons experience. But exclude eg does not affect this, we still can provide one-stop abilities for all the addons. And can solve this problem more easily.

**Comment by arkodg** [comment]
ah then lets go with the previous approach of `gateway-addons-helm` and rm `envoy-gateway` from this chart

**Comment by zhaohuabing** [comment]
Nit: should we name this directory "gateway-addons"? Currently, there are two addons(Prometheus and Grafana), and we may have more in the future.

File: charts/dashboard-addon/dashboards/envoy-clusters.json
Comment (shawnh2) [reply charts/dashboard-addon/dashboards/envoy-clusters.json:1]: done

File: charts/gateway-addon/Chart.yaml
Comment (shawnh2) [reply charts/gateway-addon/Chart.yaml]: rename to `gateway-addons-helm` by final decision

File: tools/make/helm.mk
Comment (arkodg) [inline tools/make/helm.mk:10]: can we follow similar pattern as we use for `image` ?
https://github.com/envoyproxy/gateway/blob/51a3a9dd6e71daa568db066b34cd62eae9c69185/tools/make/image.mk#L11
this will allow us to build and package individual package using commands like `make helm-package.gateway-helm` and also use top level commands like `make helm-package` to run for all charts

Comment (shawnh2) [reply tools/make/helm.mk]: done

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Comment by arkodg** [comment]
hey @phlax once this PR merges, we'll be pushing a new Helm Chart `envoyproxy/gateway-addons-helm` , will the CI be able to push using the existing PAT or do we need to create the repo before hand ?

**Review by zirain — APPROVED** [review]
LGTM, may need wait @phlax first.

**Review by guydc — APPROVED** [review]

**Pull request merged**
